salt/scripts.py

```diff
@@ -46,9 +46,24 @@ def _handle_signals(client, signum, sigframe):
         hardcrash = client.options.hard_crash
     except (AttributeError, KeyError):
         hardcrash = False
+
+    exit_msg = '\nExiting gracefully on Ctrl-c'
+    try:
+        jid = client.local_client.pub_data['jid']
```

Comment:
Good point. I personally prefer the `hasattr` approach because I feel like it's more explicit but I'm completely nitpicking. No need to change it of course. I just enjoy the discussion. ;]